theano/gpuarray/tests/test_dnn.py

```diff
@@ -2298,3 +2298,214 @@ class Cudnn_grouped_conv(Grouped_conv_noOptim):
     conv2d_gradi_op = dnn.GpuDnnConvGradI
     flip_filter = False
     is_dnn = True
+
+
+def test_dnn_spatialtf():
+    if not dnn.dnn_available(test_ctx_name):
+        raise SkipTest(dnn.dnn_available.msg)
+
+    utt.seed_rng()
+
+    """
+    Spatial Transformer implementation using Theano from Lasagne
+    Original author: skaae (https://github.com/skaae)
+    """
+    def spatialtf_cpu(inp, theta, scale_height, scale_width, border_mode='nearest'):
+        num_batch, num_channels, height, width = inp.shape
+        theta = T.reshape(theta, (-1, 2, 3))
+
+        # grid of (x_t, y_t, 1), eq (1) in ref [1]
+        out_height = T.cast(T.ceil(height * scale_height), 'int64')
+        out_width = T.cast(T.ceil(width * scale_width), 'int64')
```

Comment:
I just noticed that you call `ceil` here. But in `dnn_spatialtf()`, when you do `as_scalar(v).astype('int64')`, it's a floor that is done. That could be a problem, isnt'it ?

We should choose 1 rounding strategy and use it in both CPU and GPU functions. Maybe you should use ceil in both places. In such case, my comment about it in `dnn_spatialtf` will become:
```python
out_dims = np.ceil(out_dims).astype('int64')
```